src/CombinePDF/CombinePDF.tsx

```diff
@@ -80,6 +97,35 @@ const CombinePDF = (): JSX.Element => {
     );
   }, [doc, fileName, pagesOrder]);
 
+  const onDelete = async (pageNumber: number) => {
+    if (!doc) return;
+
+    const currentPages = new Array(doc.getPageCount())
+      .fill(0)
+      .map((_, index) => index)
+      .filter((page) => page !== pageNumber - 1);
+
+    // A document needs to at least have one page
+    // @(TODO) show error message
+    if (currentPages.length < 1) return;
+
+    const { modifiedDoc, bytes, pageCount } = await removePageFromPdf(
+      doc,
+      currentPages
+    );
+
+    const pdf = (await getDocument(bytes).promise) as PDFDocumentProxy;
+
+    const newPageOrder = new Array(pageCount)
+      .fill(0)
+      .map((_, index) => index + 1);
```

Comment:
Wouldn't this fall into the same issue you found last time? Losing the order of the current pages? Instead you can call this method by index you want to delete instead of `pageNumber` and do this:

```suggestion
    const newPageOrder = [...pagesOrder];
    newPageOrder.splice(deleteIndex, 1); // splice will mutate the array but that's ok, we already copied it.
    setPagesOrder(newPagesOrder);
```